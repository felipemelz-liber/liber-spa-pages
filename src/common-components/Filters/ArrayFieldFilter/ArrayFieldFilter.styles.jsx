import styled, { css } from 'styled-components';
import { TextField as RawField, CardFooter, Button, handleThemeFromObject } from 'liber-components';

const getTheme = (key, fallback) =>
  css`
    ${({ theme }) => handleThemeFromObject(theme, key, fallback)}
  `;

export const TextField = styled(RawField).attrs(() => ({ version: 2 }))`
  margin-bottom: 16px;
`;

export const Box = styled.div`
  width: 100%;
  height: 86px;
  padding: 0px 20px;
  display: flex;
  align-items: center;
`;

export const TagsContainer = styled.div`
  padding: 0px 20px;
  padding-bottom: 20px;
`;

export const DialogContent = styled.div`
  width: 320px;
`;

export const Footer = styled(CardFooter)`
  width: 100%;
  min-height: 52px;
  height: 52px;
  background-color: #ffffff;
  border-radius: 0px 0px 4px 4px;
  padding-top: unset;
  padding-bottom: unset;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  flex-direction: row-reverse;
`;

export const ClearButton = styled(Button).attrs(() => ({ version: '2' }))`
  font-weight: 400;
`;

export const FilterButton = styled(Button).attrs(() => ({ version: '2' }))`
  font-weight: 400;
  color: ${getTheme('colors.liber', '#009dff')};
`;
