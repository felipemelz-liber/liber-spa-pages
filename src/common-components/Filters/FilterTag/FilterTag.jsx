import React from 'react';
import PropTypes from 'prop-types';
import {
  FilterTag as Tag,
  SelectedLabelContainer,
  SelectedLabelOption,
  SelectedSwitch,
} from './FilterTag.styles';

const isSelected = list => {
  if (list) {
    return list.filter(item => Boolean(item)).length !== 0;
  }
  return false;
};

const format = list => {
  return list.join(' ');
};

const FilterTag = ({ selected, formatSelected, label, onClear, show, ...props }) => {
  const [hover, setHover] = React.useState(false);
  let tagProps;
  if (isSelected(selected)) {
    const nodeLabel = (
      <SelectedLabelContainer>
        <SelectedSwitch switch={hover || show}>
          <SelectedLabelOption>{formatSelected(selected)}</SelectedLabelOption>
          <SelectedLabelOption>{label}</SelectedLabelOption>
        </SelectedSwitch>
      </SelectedLabelContainer>
    );
    tagProps = {
      label: nodeLabel,
      closable: true,
      onClose: onClear,
      onMouseEnter: () => setHover(true),
      onMouseLeave: () => setHover(false),
      gray: !show,
      show,
      ...props,
    };
  } else {
    tagProps = { label, show, ...props };
  }
  return <Tag {...tagProps} />;
};

FilterTag.propTypes = {
  selected: PropTypes.arrayOf(PropTypes.any),
  formatSelected: PropTypes.func,
  onClear: PropTypes.func,
  label: PropTypes.string,
  show: PropTypes.bool,
};

FilterTag.defaultProps = {
  selected: [],
  formatSelected: format,
  onClear: () => {},
  show: false,
  label: '',
};

export default FilterTag;
