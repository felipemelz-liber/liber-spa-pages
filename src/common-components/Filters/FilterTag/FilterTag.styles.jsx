import styled, { css } from 'styled-components';
import { ModalTag } from 'liber-components';

const isGray = ({ gray }) => {
  if (gray)
    return css`
      background-color: #f4f5f7;
      border-color: #f4f5f7;
    `;
  return null;
};

export const FilterTag = styled(ModalTag)`
  font-size: 16px;
  min-height: 36px;
  border-radius: 4px;
  font-weight: 400;
  ${isGray}
`;

export const SelectedLabelContainer = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
  overflow: hidden;
`;

export const SelectedSwitch = styled.div`
  height: 52px;
  width: 100%;
  padding-top: 6px;
  display: flex;
  flex-direction: column;
  transform: ${props => (props.switch ? 'translateY(-26px)' : 'translateY(0px)')};
  transition: transform 250ms ease-in-out;
`;

export const SelectedLabelOption = styled.span`
  width: 100%;
  height: 26px;
  display: flex;
  justify-content: center;
  align-items: center;
`;
