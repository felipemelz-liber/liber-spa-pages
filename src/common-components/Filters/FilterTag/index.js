export { default } from './FilterTag.jsx';
