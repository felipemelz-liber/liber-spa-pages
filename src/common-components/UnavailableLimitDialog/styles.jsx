import styled from 'styled-components';
import { FlatButton } from 'liber-components';

const TEXT_COLOR = '#6287a7';

export const Container = styled.div`
  padding: 24px 24px 12px 24px;
  width: 480px;

  * {
    font-family: Roboto;
  }
`;

export const Header = styled.div`
  font-size: 21px;
  font-weight: bold;
  line-height: 1.4;
  letter-spacing: 0.08px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const Content = styled.div`
  font-size: 16px;
  line-height: 1.38;
  letter-spacing: 0.01px;
  margin-bottom: 24px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const ActionsRow = styled.div`
  display: flex;
  justify-content: flex-end;
`;

export const Button = styled(FlatButton).attrs(() => ({ size: 'large' }))``;
