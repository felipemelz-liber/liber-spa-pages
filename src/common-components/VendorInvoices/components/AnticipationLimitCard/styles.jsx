import styled from 'styled-components';

export const Card = styled.div`
  font-family: Roboto;
  color: #6287a7;
  width: 360px;
  box-shadow: 0px 3px 6px #00000029;
  border: 1px solid #cdd7df;
  border-radius: 8px;
  padding: 16px 24px;
  margin-bottom: 56px;

  * {
    font-family: Roboto;
  }
`;

export const CardTitle = styled.div`
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.06px;
  font-weight: 500;
  color: #4b6f85;
  margin-bottom: 4px;
`;
