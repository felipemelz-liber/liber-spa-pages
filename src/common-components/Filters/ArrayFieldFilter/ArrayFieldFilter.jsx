import React, { useState } from 'react';
import PropTypes from 'prop-types';
import FilterTag from '../FilterTag/index.js';
import {
  TextField,
  Box,
  TagsContainer,
  FilterButton,
  ClearButton,
  Footer,
  DialogContent,
} from './ArrayFieldFilter.styles.jsx';
import TagsDisplay from './components/TagsDisplay';

const toggleOpen = setOpen => value => {
  setOpen(value);
};

const hasSeparator = text => text.includes(',') || text.includes(';') || text.includes(' ');

const splitText = text => {
  const spacesCleared = text.replace(/\s/g, ',');
  const withCommaSeparator = spacesCleared.replace(/[;]+/g, ',');
  return withCommaSeparator.split(',');
};

const cleanArray = items => items.filter(item => item !== '').map(item => item.trim());

const onConfirm = (textState, onFilter, setOpen) => event => {
  const [text, setText] = textState;
  let values;
  event.preventDefault();

  if (hasSeparator(text)) {
    const items = splitText(text);
    values = cleanArray(items);
  } else {
    values = [text.trim()];
  }

  setOpen(false);
  setText('');
  onFilter(values);
};

const formatSelectedLabel = label => list => {
  const [selectedItems] = list;
  return `${label} (${selectedItems.length})`;
};

const ArrayFieldFilter = ({
  onClear,
  onFilter,
  selectedItems,
  hintText,
  label,
  tagsThreshold,
  tagsPerLine,
}) => {
  const textState = useState('');
  const [open, setOpen] = useState(false);
  const [text, setText] = textState;

  return (
    <FilterTag
      selected={[selectedItems]}
      label={label}
      show={open}
      onChange={toggleOpen(setOpen)}
      onClear={onClear}
      formatSelected={formatSelectedLabel(label)}
    >
      <form>
        <DialogContent>
          <Box>
            <TextField autoFocus hintText={hintText} width="100%" value={text} onChange={setText} />
          </Box>
          <TagsContainer>
            <TagsDisplay
              tags={selectedItems || []}
              tagsThreshold={tagsThreshold}
              tagsPerLine={tagsPerLine}
            />
          </TagsContainer>
        </DialogContent>
        <Footer>
          <FilterButton disabled={text === ''} onClick={onConfirm(textState, onFilter, setOpen)}>
            FILTRAR
          </FilterButton>
          <ClearButton onClick={onClear} disabled={!selectedItems || selectedItems.length === 0}>
            LIMPAR
          </ClearButton>
        </Footer>
      </form>
    </FilterTag>
  );
};

ArrayFieldFilter.propTypes = {
  selectedItems: PropTypes.arrayOf(PropTypes.any),
  onFilter: PropTypes.func,
  onClear: PropTypes.func,
  hintText: PropTypes.string,
  tagsThreshold: PropTypes.number,
  label: PropTypes.string,
  tagsPerLine: PropTypes.number,
};

ArrayFieldFilter.defaultProps = {
  selectedItems: null,
  hintText: '',
  tagsThreshold: 18,
  tagsPerLine: 6,
  onFilter: () => {},
  onClear: () => {},
  label: '',
};

export default ArrayFieldFilter;
