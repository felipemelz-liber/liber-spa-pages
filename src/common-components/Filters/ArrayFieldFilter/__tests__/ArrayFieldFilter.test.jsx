import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import ArrayFieldFilter from '../ArrayFieldFilter';
import { ClearButton, FilterButton, TextField } from '../ArrayFieldFilter.styles';
import FilterTag from '../../FilterTag';

describe('ArrayFieldFilter tests', () => {
  configure({ adapter: new Adapter() });

  const props = {
    label: 'mock',
    selectedItems: ['mock', 'mock', 'mock'],
    onFilter: sinon.spy(),
    onClear: sinon.spy(),
  };

  const event = { preventDefault: sinon.spy() };

  afterEach(() => {
    const { onFilter, onClear } = props;
    const { preventDefault } = event;
    onFilter.resetHistory();
    onClear.resetHistory();
    preventDefault.resetHistory();
  });

  it('should render correctly without props', () => {
    const wrapper = shallow(<ArrayFieldFilter />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly with props', () => {
    const wrapper = shallow(<ArrayFieldFilter {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call onFilter when filter Button is clicked', () => {
    const wrapper = shallow(<ArrayFieldFilter {...props} />);
    wrapper.find(FilterButton).simulate('click', event);

    const { onFilter } = props;
    expect(event.preventDefault.called).toBeTruthy();
    expect(onFilter.called).toBeTruthy();
  });

  it('should call onClear when clear button is clicked', () => {
    const wrapper = shallow(<ArrayFieldFilter {...props} />);
    wrapper.find(ClearButton).simulate('click');
    const { onClear } = props;
    expect(onClear.called).toBeTruthy();
  });

  it('should call onClear when FilterTagClears', () => {
    const wrapper = shallow(<ArrayFieldFilter {...props} />);
    wrapper.find(FilterTag).simulate('clear');
    const { onClear } = props;
    expect(onClear.called).toBeTruthy();
  });

  it('should call onFilter with correct values', () => {
    const wrapper = shallow(<ArrayFieldFilter {...props} />);

    wrapper.find(TextField).simulate('change', 'mock, mock; mock mock');
    wrapper.find(FilterButton).simulate('click', event);

    const { onFilter } = props;
    expect(event.preventDefault.called).toBeTruthy();
    expect(onFilter.calledWith(['mock', 'mock', 'mock', 'mock'])).toBeTruthy();
  });
});
