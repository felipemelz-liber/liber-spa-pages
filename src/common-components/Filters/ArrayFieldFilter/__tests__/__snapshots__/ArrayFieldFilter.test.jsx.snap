// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrayFieldFilter tests should render correctly with props 1`] = `
<FilterTag
  formatSelected={[Function]}
  label="mock"
  onChange={[Function]}
  onClear={[Function]}
  selected={
    Array [
      Array [
        "mock",
        "mock",
        "mock",
      ],
    ]
  }
  show={false}
>
  <form>
    <styled.div>
      <styled.div>
        <Styled(Component)
          autoFocus={true}
          hintText=""
          onChange={[Function]}
          value=""
          width="100%"
        />
      </styled.div>
      <styled.div>
        <TagsDisplay
          tags={
            Array [
              "mock",
              "mock",
              "mock",
            ]
          }
          tagsPerLine={6}
          tagsThreshold={18}
        />
      </styled.div>
    </styled.div>
    <Styled(Cards__CardFooter)>
      <Styled(Buttons__Button)
        disabled={true}
        onClick={[Function]}
      >
        FILTRAR
      </Styled(Buttons__Button)>
      <Styled(Buttons__Button)
        disabled={false}
        onClick={[Function]}
      >
        LIMPAR
      </Styled(Buttons__Button)>
    </Styled(Cards__CardFooter)>
  </form>
</FilterTag>
`;

exports[`ArrayFieldFilter tests should render correctly without props 1`] = `
<FilterTag
  formatSelected={[Function]}
  label=""
  onChange={[Function]}
  onClear={[Function]}
  selected={
    Array [
      null,
    ]
  }
  show={false}
>
  <form>
    <styled.div>
      <styled.div>
        <Styled(Component)
          autoFocus={true}
          hintText=""
          onChange={[Function]}
          value=""
          width="100%"
        />
      </styled.div>
      <styled.div>
        <TagsDisplay
          tags={Array []}
          tagsPerLine={6}
          tagsThreshold={18}
        />
      </styled.div>
    </styled.div>
    <Styled(Cards__CardFooter)>
      <Styled(Buttons__Button)
        disabled={true}
        onClick={[Function]}
      >
        FILTRAR
      </Styled(Buttons__Button)>
      <Styled(Buttons__Button)
        disabled={true}
        onClick={[Function]}
      >
        LIMPAR
      </Styled(Buttons__Button)>
    </Styled(Cards__CardFooter)>
  </form>
</FilterTag>
`;
